cmd/cloud-controller-manager/controller-manager.go

```diff
@@ -51,6 +54,7 @@ func main() {
 	defer logs.FlushLogs()
 
 	if err := command.Execute(); err != nil {
-		os.Exit(1)
+		exitCode = 1
```

Comment:
Sadly this doesn't work (the os.Exit part with this) because the deferred functions parameters are evaluated immediately. See https://play.golang.org/p/MuR_9TV-twr .  This also demonstrated that a referenced variable rather than passed parameter can work. (Beware using this in a goroutine as its very racy)

Reply:
Yes buddy, you are right.

Reply:
Call `os.Exit(exitCode)` in anonymous function should work.

I udpate the code , and here is an example:
```
package main

import (
	"os"
)

func main() {
	exitCode := 0
	defer func() { os.Exit(exitCode) }()
	exitCode = 1
}
```